scripts/lib/CIME/XML/compilerblock.py

```diff
@@ -190,8 +190,7 @@ def _elem_to_setting(self, elem):
         # Create the setting object.
         append = self._db.name(elem) == "append" or (self._db.name(elem) == "base" and self._compiler and self._db.compiler != self._compiler)
```

Comment:
I know that this `force_no_append` stuff was already merged, but I've decided that I really don't like it. I think that the real problem is this line, which makes no sense at all. As far as I can tell, this line is trying to get some `<base>` tags to be treated as `<append>` tags in order to trick an ambiguity checker down the line. But when they were ending up actually being appended we were forced to put in this `force_no_append` bit to work around that.

The right answer would be to change this line to simply:

    append = self._db.name(elem) == "append"

Then remove the `force_no_append` flag, and if that causes problems with the ambiguity checker, then that code should be patched properly!